src/Apps/Auction/Components/RegistrationForm.tsx

```diff
@@ -260,41 +280,70 @@ const RegistrationForm: React.FC<RegistrationFormProps> = props => {
     })
   }
 
-  return (
-    <Formik
-      initialValues={{
-        name: "",
-        street1: "",
-        city: "",
-        credit_card: null,
-        state: "",
-        postal_code: "",
-        telephone: "",
-        agree_to_terms: false,
-      }}
-      onSubmit={(values: FormValues, { setFieldError, setSubmitting }) => {
-        const address = {
-          name: values.name,
-          address_line1: values.street1,
-          address_city: values.city,
-          address_state: values.state,
-          address_zip: values.postal_code,
-        }
+  const [showErrorModal, setShowErrorModal] = useState(false)
 
-        props.stripe.createToken(address).then(({ error, token }) => {
-          if (error) {
-            setFieldError("credit_card", error.message)
-            setSubmitting(false)
-          } else {
-            createCreditCard(token.id).then(() => {
-              createBidder(setSubmitting)
-            })
+  return (
+    <>
+      <Formik
+        initialValues={{
+          name: "",
+          street1: "",
+          country: "",
+          city: "",
+          credit_card: null,
+          state: "",
+          postal_code: "",
+          telephone: "",
+          agree_to_terms: false,
+        }}
+        onSubmit={(values: FormValues, actions) => {
```

Comment:
have you considered extracted a named function here? Maybe this is idiomatic TS and I'm not used to it, but seems like this file might be a bit easier to read by passing in and naming a function

Reply:
Yep :+1:  As hinted to in https://github.com/artsy/reaction/pull/2703#discussion_r315741182, I think we're missed a component level here that should clean this up.